### PR TITLE
[cmake] Allow building with Ninja generator

### PR DESCRIPTION
**Pull request by fetzerch**
Small fixes to enable building with Ninja. Tested on OSX and Linux.
The build with Ninja is only a little faster (couple of seconds for a full build), but Ninja has a few nice things like generating a dependency graph.

Might be interesting for: @hudokkow, @notspiff, @wsnipex.

### CODE REVIEW

**Comment by fetzerch** [comment]
@AchimTuran: Works also on Windows now. I remember that you once tried that and ran into some issues. If you like, give it another try :).

File: project/cmake/modules/FindCrossGUID.cmake [hunk 1]
```diff
@@ -18,7 +18,8 @@ if(ENABLE_INTERNAL_CROSSGUID)
                                     <SOURCE_DIR> &&
                                     ${CMAKE_COMMAND} -E copy
                                     ${CORE_SOURCE_DIR}/tools/depends/target/crossguid/FindCXX11.cmake
-                                    <SOURCE_DIR>)
+                                    <SOURCE_DIR>
+                      BUILD_BYPRODUCTS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/libcrossguid.a)
```
Comment (wsnipex) [inline project/cmake/modules/FindCrossGUID.cmake]: maybe use a var for the lib here and in CROSSGUID_LIBRARIES below?
Same for dvd libs

Comment (fetzerch) [inline project/cmake/modules/FindCrossGUID.cmake]: Oh sure, indeed.

**Comment by AchimTuran** [comment]
Awesome. Thanks I will test it during the week. I guess it could speed up our jenkins builds.

**Comment by fetzerch** [comment]
Thanks @AchimTuran for testing, I forgot the Effects11 lib (didn't clean my source dir, hence I didn't get any error).
Now it's working also on Windows.

**Pull request merged**